src/etc/inc/certs.inc

```diff
@@ -1652,13 +1644,29 @@ function cert_renew(& $cert, $reusekey = true, $strictsecurity = false) {
 	$args = array(
 		"x509_extensions" => $cert_type,
 		"digest_alg" => $digest_alg,
-		"private_key_bits" => (int)$private_key_bits,
 		"private_key_type" => $private_key_type,
 		"encrypt_key" => false);
 
-	/* Adjust argumnents to account for ECDSA keys. */
-	if (($key_details['type'] ==  OPENSSL_KEYTYPE_EC) &&
-	    (!empty($key_details['ec']['curve_name']))) {
+	/* If we are enforcing strict security, then ensure the private key size
+	 * is at least 2048 bits or NIST P-256 elliptic curve*/
+	$private_key_bits = $key_details['bits'];
+	if ($strictsecurity) {
+		if (($key_details['type'] == OPENSSL_KEYTYPE_RSA) &&
+		    ($private_key_bits < $cert_strict_values['min_private_key_bits'])) {
+			$private_key_bits = $cert_strict_values['min_private_key_bits'];
+			$reusekey = false;
+		} else if (!in_array($key_details['ec']['curve_name'], $cert_curve_compatible['IPsec'])) {
```

Comment:
Rather than arbitrarily choosing `IPsec` as the list here, we can check it against all known compatible values using something like `array_unique(call_user_func_array('array_merge', $cert_curve_compatible))` to get the full list of compatible curves for any purpose.

At the moment the lists may be similar but that may not always be the case in the future.

Reply:
done